doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +108,354 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
+
+* :n:`@ident` - use the specified name
+* :n:`?` - let Coq choose a name
+* :n:`?@ident` - generate a name that begins with :n:`@ident`
+* :n:`_` - discards the matched part (unless it is required for another
+  hypothesis)
+* if a disjunction pattern omits a name, such as :g:`[|H2]`, Coq will choose a name
+
+**Splitting patterns**
+
+The most common splitting patterns are:
+
+* split a hypothesis in the form :n:`A /\ B` into two
+  hypotheses :g:`H1: A` and :g:`H2: B` using the pattern :g:`(H1 & H2)` or
+  :g:`(H1, H2)` or :g:`[H1 H2]`
+  :ref:`Example <intropattern_conj_ex>`
+* split a hypothesis in the form :g:`A \/ B` into two
+  subgoals using the pattern :g:`[H1|H2]`.  The first subgoal will have the hypothesis
+  :g:`H1: A` and the second subgoal will have the hypothesis :g:`H2: B`.
+  :ref:`Example <intropattern_disj_ex>`
+* split a hypothesis in either of the forms :g:`A /\ B` or :g:`A \/ B` using the pattern :g:`[]`.
+
+**A node has to be of type Prop to be matched/split/named, right?**
+
+**what other common operators are intro patterns useful for?  ->? <->? forall?  I would like to
+mention them**
+
+Patterns can be nested: :n:`[[Ha|Hb] H]` can be used to split :n:`(A \/ B) /\ C`.
+
+Note that there is no equivalent to intro patterns for goals.  For a goal :g:`A /\ B`,
+use the :tacn:`split` tactic to replace the current goal with subgoals :g:`A` and :g:`B`.
+For a goal :g:`A \/ B`, use :tacn:`left` to replace the current goal with :g:`A`, or
+:tacn:`right` to replace the current goal with :g:`B`.
+
+* **() - I couldn't figure out how this worked.  I think it did something for intros (lost the example) but
+  it wasn't valid for any destruct I tried.  Is it actually useful?**
+
+* :n:`( {+, @simple_intropattern}` ) - matches
+  a product over an inductive type with a
+  :ref:`single constructor <intropattern_cons_note>`.
+  If the number of patterns
+  equals the number of constructor arguments, then it applies the patterns only to
+  the arguments, and
+  :n:`( {+, @simple_intropattern} )` is equivalent to :n:`[{+ @simple_intropattern}]`.
+  If the number of patterns equals the number of constructor arguments plus the number
+  of :n:`let-ins`, the patterns are applied to the arguments and :n:`let-in` variables.
+
+* :n:`( {+& @simple_intropattern} )` - matches a term that consists of one or more
+  right-hand nested binary inductive types in a form such as :g:`(a1 OP (a2 OP ...))`.
+  The splitting pattern can have more than 2 names, for example :g:`(H1 & H2 & H3)`
+  matches :g:`A /\ B /\ C`.
+  The inductive types must have a
+  :ref:`single constructor with two parameters <intropattern_cons_note>`.
+  :ref:`Example <intropattern_ampersand_ex>`
+
+* :n:`[ {+| @intropatterns} ]` - splits an inductive type that has
+  :ref:`multiple constructors <intropattern_cons_note>`
+  such as :n:`A \/ B`
+  into multiple subgoals.  The number of ``intropatterns`` must be the same as the number of
```

Comment:
```suggestion
  into multiple subgoals.  The number of :token:`intropatterns` must be the same as the number of
```

Reply:
FWIW It's awkward to have both `intropattern` and `intropatterns` non-terminals--makes it hard to distinguish singular and plural references.

Reply:
I think we should banish plural non-terminal names like `intropatterns` in favor of `intropattern_list`.  But not in this PR.

Reply:
Well, I agree at least that as part of this PR, the token be named `introppatern_list` instead of `intropatterns`.

Reply:
Done, but only in the doc.